grid/datasourcerequest-on-server/CustomSerializer/Server/Controllers/WeatherForecastController.cs

```diff
@@ -0,0 +1,80 @@
+﻿using CustomSerializer.Shared;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Telerik.DataSource;
+using Telerik.DataSource.Extensions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace CustomSerializer.Server.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class WeatherForecastController : ControllerBase
+    {
+        private static readonly string[] Summaries = new[]
+        {
+                "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
+            };
+
+        private readonly ILogger<WeatherForecastController> logger;
+
+        public WeatherForecastController(ILogger<WeatherForecastController> logger)
+        {
+            this.logger = logger;
+        }
+
+        // this static list acts as our "database" in this sample
+        private static List<WeatherForecast> _forecasts { get; set; }
+
+        [HttpPost]
+        public async Task<DataEnvelope<WeatherForecast>> Post([FromBody] DataSourceRequest request)
+        {
+            // generate some data for the sake of this demo
+            if (_forecasts == null)
+            {
+                var rng = new Random();
+                var startDate = DateTime.Now.Date;
+                _forecasts = Enumerable.Range(1, 150).Select(index => new WeatherForecast
+                {
+                    Id = index,
+                    Date = startDate.AddDays(index),
+                    TemperatureC = rng.Next(-20, 55),
+                    Summary = Summaries[rng.Next(Summaries.Length)]
+                }).ToList();
+            }
+
+            // we will cast the data to an IQueriable to simulate an actual database (EF) service
+            // in a real case, you would be fetching the data from the service, not generating it here
+            IQueryable<WeatherForecast> queriableData = _forecasts.AsQueryable();
+
+            // deserialize the DataSourceRequest from the HTTP query if it comes as a string like
+            // Post([FromBody]string dsRequest)
+            // for example, here's how this could look like with the default System.Text.Json serializer
+            // in this example, it comes in as a type parameter that the framework should deserialize for you
+            // which is catered for by the custom converter that is registered to Newtsonsoft in Startup.cs
+            //DataSourceRequest gridRequest = JsonSerializer.Deserialize<DataSourceRequest>(dsRequest);
```

Comment:
I think this section is a bit confusing as we directly receive the DataSourceRequest, and while reading I found it a bit stange and out of context.